index.js

```diff
@@ -14,14 +14,25 @@ var builder = require('./lib/builder');
 
 var path = require('path');
 
-/* Very basic, null-friendly, shallow clone for object attributes only */
-function clone(obj) {
-  obj = obj || {};
-  var copy = {};
-  for (var key in obj) {
-    copy[key] = obj[key];
-  }
-  return copy;
+function processOpts(opts_, outFile) {
+  var keys = [
+    'config',
+    'lowResSourceMaps',
+    'minify',
+    'normalize',
+    'outFile',
+    'runtime',
+    'sourceMaps',
+    'sourceMapContents'
+  ];
+  var opts = {};
+  keys.forEach(function(key) {
+    opts[key] = opts_ && opts_[key];
+  });
```

Comment:
The point of this is to allow type inference through the setting of default values.

A common pattern I find useful is:

```javascript
function prepend(a, b) {
  for (var p in b) {
    if (!a[p])
      a[p] = b[p];
  }
  return a;
}

function processOpts(opts_, outFile) {
  // apply defaults
  prepend(opts_, {
    config: {},
    lowResSourceMaps: true,
    minify: true,
    normalize: false,
    outFile: '',
    runtime: false,
    sourceMaps: true,
    sourceMapContents: false
  });
}
```

Sorry to be a pain...